tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -247,6 +284,13 @@ pub mod pallet {
 			IDGraphs::<T>::try_mutate(&who, &identity, |context| -> DispatchResult {
 				let mut c = context.take().ok_or(Error::<T>::IdentityNotExist)?;
 
+				if c.metadata.is_none()
+					&& c.creation_request_block == Some(0)
+					&& c.verification_request_block == Some(0)
+				{
+					ensure!(!c.is_verified, Error::<T>::IdentityAlreadyVerified);
+				}
+
```

Comment:
I think this should be extended to non-prime Identities as well - we should quite early if it's already verified.